Lib/pdb.py

```diff
@@ -514,35 +514,54 @@ def displayhook(self, obj):
         if obj is not None:
             self.message(repr(obj))
 
+    @contextmanager
+    def _disable_tab_completion(self):
+        if self.use_rawinput and self.completekey == 'tab':
+            try:
+                import readline
+            except ImportError:
+                yield
+                return
+
+            if self.use_rawinput and self.completekey == 'tab':
+                try:
+                    readline.parse_and_bind('tab: self-insert')
+                    yield
+                finally:
+                    readline.parse_and_bind('tab: complete')
+            else:
+                yield
```

Comment:
I like the context manager approach, but I'm confused by the logic. I think you have an extra `if self.use_rawinput and self.completekey == 'tab':` here, right?

```suggestion
        if self.use_rawinput and self.completekey == 'tab':
            try:
                import readline
            except ImportError:
                yield
                return
            try:
                readline.parse_and_bind('tab: self-insert')
                yield
            finally:
                readline.parse_and_bind('tab: complete')
        else:
            yield
```

Reply:
You are right, that was a mistake when I was trying to figure out the logic of the context manager.